network-core/src/client/gossip.rs

```diff
@@ -0,0 +1,13 @@
+use super::Error;
+
+use chain_core::property::{Gossip, NodeId};
+
+use futures::prelude::*;
+
+pub trait GossipService<T: Gossip> {
```

Comment:
Don't use a generic parameter here. `BlockService` was wrong about this, #486 fixes that.